editor/modes/visual-editor/block-drop-zone.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External Dependencies
+ */
+import { connect } from 'react-redux';
+import { reduce, get, find } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { DropZone } from '@wordpress/components';
+import { getBlockTypes } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { insertBlocks } from '../../actions';
+
+function BlockDropZone( { index, ...props } ) {
+	const dropFiles = ( files, position ) => {
+		const transformation = reduce( getBlockTypes(), ( ret, blockType ) => {
+			if ( ret ) {
+				return ret;
+			}
+
+			return find( get( blockType, 'transforms.from', [] ), ( transform ) => (
+				transform.type === 'files' && transform.isMatch( files )
+			) );
+		}, false );
+
+		if ( transformation ) {
+			let insertPosition;
+			if ( index !== undefined ) {
+				insertPosition = position.y === 'top' ? index : index + 1;
+			}
+			transformation.transform( files ).then( ( blocks ) => {
```

Comment:
I'm not saying we need an alternative necessarily, just to consider:

- Could it ever be synchronous? Do we want to support this?
- Impact of consistency between other transform return values